src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/eye/eye_kernel_ref.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "eye_kernel_ref.h"
+
+#include <kernel_selector_utils.h>
+
+#include <vector>
+
+namespace {
+
+void addJitConstantForAttribute(kernel_selector::JitConstants& jit,
+                                const std::string& name,
+                                const std::int32_t attribute) {
+    using namespace kernel_selector;
+    jit.AddConstant(MakeJitConstant(name, attribute));
+}
+
+}  // anonymous namespace
+
+namespace kernel_selector {
+
+KernelsData EyeKernelRef::GetKernelsData(const Params& params, const optional_params& options) const {
+    if (!Validate(params, options)) {
+        return {};
+    }
+    KernelData kernel_data = KernelData::Default<eye_params>(params);
+    const eye_params& new_params = dynamic_cast<eye_params&>(*kernel_data.params.get());
+    auto dispatch_data = SetDefault(new_params, options);
+    auto entry_point = GetEntryPoint(kernelName, new_params.layerID, params, options);
+    auto slice_specific_jit = GetJitConstants(new_params);
+    auto jit = CreateJit(kernelName, slice_specific_jit, entry_point);
+
+    FillCLKernelData(kernel_data.kernels[0], dispatch_data, params.engineInfo, kernelName, jit, entry_point);
+    return {kernel_data};
+}
+
+KernelsPriority EyeKernelRef::GetKernelsPriority(const Params& /*params*/, const optional_params& /*options*/) const {
+    return DONT_USE_IF_HAVE_SOMETHING_ELSE;
+}
+
+ParamsKey EyeKernelRef::GetSupportedKey() const {
+    ParamsKey k;
+    k.EnableInputDataType(Datatype::INT32);
+    k.EnableInputDataType(Datatype::INT64);
+    k.EnableInputDataType(Datatype::F32);
```

Comment:
Input data type can be f32? Because in doc I see that it can be only int32 or int64.

Reply:
added comment why it is necessary